crates/bevy_render/src/mesh/primitives/dim3/cylinder.rs

```diff
@@ -0,0 +1,184 @@
+use bevy_math::primitives::Cylinder;
+use wgpu::PrimitiveTopology;
+
+use crate::{
+    mesh::{Indices, Mesh, Meshable},
+    render_asset::RenderAssetUsages,
+};
+
+/// A builder used for creating a [`Mesh`] with a [`Cylinder`] shape.
+#[derive(Clone, Copy, Debug)]
+pub struct CylinderMeshBuilder {
+    /// The [`Cylinder`] shape.
+    pub cylinder: Cylinder,
+    /// The number of vertices used for the top and bottom of the cylinder.
+    ///
+    /// The default is `32`.
+    pub resolution: u32,
```

Comment:
I think `resolution` and `segments` are a bit unclear. Perhaps `circle_resolution` and `height_segments` or something?

Reply:
These are the same names as before. Ironically, I specifically didn't change these to avoid making this more controversial.

`circle_resolution` could be fine, or maybe `base_resolution`... Same for `height_segments`, but it could also be `vertical_segments`, `vertical_resolution`, or `rings`, which is what the capsule uses. I'd probably go for `rings` for the sake of consistency, but I'm not entirely satisfied with any of these names, so personally I'd leave it like this for now and consider renaming in a follow-up.

I can rename if others have preferences though

Reply:
I would prefer to do any name changes after we switch the renderer over. One less thing to worry about.